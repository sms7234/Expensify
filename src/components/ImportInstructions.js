import React from 'react';
import Accordion from 'react-bootstrap/Accordion';
import Card from 'react-bootstrap/Card';

export default (props) => (
  <div className="content-container--subHeader">
    <Accordion>
      <Card>
        <Accordion.Toggle
          as={Card.Header}
          className ="accordion__header"
          eventKey="0">
          How to instructions
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="0">
          <Card.Body className="content-container accordion__message">
            <h3>Step 1: Setup & Save Local CSV file</h3>
            <p>Be sure the csv file has the following headers: </p>
            <ul>
              {props.dataFormat.map((item,index) => {return <li key={index}>{item}</li>})}
            </ul>
            <h3>Step 2: Upload the .csv file</h3>
            <h3>Step 3: Click import</h3>
            <h3>Step 4: Ensure integrity of information by updating the populated fields</h3>
            <ul>
              <li>All fieds must be completed, with the exception of note</li>
              <li>Any edits on an expense item must be saved before moving to the next line item.  Otherwise the changes will be lost</li>
              <li>Click on "Check Data" button to ensure all fields are correctly populated.  Be sure to update & save any line items that are highlighted in red.</li>
            </ul>
            <h3>Step 5: Once all line items are green click the submit button</h3>
          </Card.Body>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  </div>
)
