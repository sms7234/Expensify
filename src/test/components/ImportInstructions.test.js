import React from 'react';
import {shallow} from 'enzyme';
import ImportInstructions from '../../components/ImportInstructions';

test('match snapshot of instructions page', () => {
  const wrapper = shallow(<ImportInstructions />);
  expect(wrapper).toMatchSnapshot();
})
