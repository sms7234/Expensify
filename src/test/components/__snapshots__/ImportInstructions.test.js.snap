// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot of instructions page 1`] = `
<div
  className="content-container--subHeader"
>
  <ForwardRef>
    <Card
      body={false}
    >
      <ForwardRef
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "CardHeader",
            "render": [Function],
          }
        }
        className="accordion__header"
        eventKey="0"
      >
        How to instructions
      </ForwardRef>
      <AccordionCollapse
        eventKey="0"
      >
        <CardBody
          className="content-container accordion__message"
        >
          <h3>
            Step 1: Setup & Save Local CSV file
          </h3>
          <p>
            Be sure the csv file has the following headers: 
          </p>
          <ul>
            <li>
              Date (MM/DD/YY)
            </li>
            <li>
              Amount
            </li>
            <li>
              Category
            </li>
            <li>
              Business
            </li>
            <li>
              Note
            </li>
          </ul>
          <h3>
            Step 2: Upload the .csv file
          </h3>
          <h3>
            Step 3: Click import
          </h3>
          <h3>
            Step 4: Ensure integrity of information by updating the populated fields
          </h3>
          <ul>
            <li>
              All fieds must be completed, with the exception of note
            </li>
            <li>
              Any edits on an expense item must be saved before moving to the next line item.  Otherwise the changes will be lost
            </li>
            <li>
              Click on "Check Data" button to ensure all fields are correctly populated.  Be sure to update & save any line items that are highlighted in red.
            </li>
          </ul>
          <h3>
            Step 5: Once all line items are green click the submit button
          </h3>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </ForwardRef>
</div>
`;
